app/src/main/java/co/develhope/chooseyouowncocktail_g2/DrinkList.kt

```diff
@@ -31,6 +33,32 @@ object DrinkList {
             drink,
             drink.copy(favourite = bool)
         )
+
+    }
+
+    fun booleanSortDrinkList(){
+        drinks = this.drinks.sortedBy { it.favourite }
+        drinks.forEach{ Log.d("debug", "${it.name} and ${it.favourite}") }
+
+    }
+
+    fun returnOnlyPreferiteSelectedDrink(): List<Drink> {
+        return drinks.filter { it.favourite == true }
     }
 
+    fun repleacePreferiteOnSelfTop(preferiteList: List<Drink>){
+        drinks = drinks.toMutableList()
+        for(prefDrink in preferiteList){
+            (drinks as MutableList<Drink>).remove(prefDrink)
+        }
+        drinks.forEach { Log.d("debug2", "${it.name} and ${it.favourite}")}
+
+        val tmpDrinkList = ArrayList<Drink>()
+        tmpDrinkList.addAll(preferiteList)
```

Comment:
Avrei lasciato il reversed(), in modo che l'ultimo preferito aggiunto sia sempre in cima